service/create/service.go

```diff
@@ -824,7 +842,7 @@ func (s *Service) addFunc(obj interface{}) {
 	// If the policy couldn't be created and some instances didn't exist before, that means that the cluster
 	// is inconsistent and most problably its deployment broke in the middle during the previous run of
 	// aws-operator.
-	if anyMastersCreated && (kmsKeyErr != nil || policyErr != nil) {
+	if anyMastersCreated && (kmsKeyErr != nil || masterPolicyErr != nil || workerPolicyErr != nil) {
```

Comment:
I am wondering why we do not handle all these errors separately. I think these errors should be handled somehow even if there are no masters created. I also do not see where these errors come from, so this could be clearer. I am not into the details here but I think errors should immediately handled when they occur. This might be a candidate for improvements when we move to the operatorkit reconciliation framework.